cve_bin_tool/checkers/__init__.py

```diff
@@ -96,6 +96,14 @@ def __new__(cls, name, bases, props):
         cls.VENDOR_PRODUCT = list(
             map(lambda vpkg: VendorProductPair(*vpkg), cls.VENDOR_PRODUCT)
         )
+        # Validate that vendor product pair is in lowercase
+        for items in cls.VENDOR_PRODUCT:
+            for vp in items:
+                if vp.islower() != True:
+                    raise AssertionError(
```

Comment:
Mostly we log errors but don't fail in cve-bin-tool, but the use case for this is "developer added new checker that has a flaw" so we probably do want to halt with an error.  So I think raising an error is correct, but assertions in python are a bit of a problem (they get optimized out of production code, so they're used sparingly in test code only) so it probably shouldn't be `AssertionError` that's raised.  

I think in other places we've made our own errors, so probably you want to make something like InvalidCheckerError and raise that instead.  Take a look at https://github.com/intel/cve-bin-tool/blob/master/cve_bin_tool/error_handler.py to see what the others look like, and probably add it there.